### PR TITLE
Chain stream instead of merge

### PR DESCRIPTION
**Pull request by RDruon**
Chain stream instead of merging them.

Merge will interleave stream, that's not something we want since it will break prometheus format. See merge documentation at https://docs.rs/tokio-stream/latest/tokio_stream/trait.StreamExt.html#method.merge

Also add a newline to make sure we are printing stats with a separating empty line

From (mind the `{operation="open",component="mdt",target="es01a-MDT0000",jobid="node_exporter:989:es-2-virt3-co"}`):
```
[root@node ~]# curl http://localhost:32221/metrics
....
lustre_stats_total{component="mdt",operation="sync",target="es02a-MDT0000"} 6
lustre_stats_total{component="mdt",operation="samedir_rename",target="es02a-MDT0000"} 2
lustre_stats_total{component="mdt",operation="parallel_rename_file",target="es02a-MDT0000"} 2
{operation="open",component="mdt",target="es01a-MDT0000",jobid="node_exporter:989:es-2-virt3-co"} 0
lustre_job_stats_total{operation="close",component="mdt",target="es01a-MDT0000",jobid="node_exporter:989:es-2-virt3-co"} 0
lustre_job_stats_total{operation="mknod",component="mdt",target="es01a-MDT0000",jobid="node_exporter:989:es-2-virt3-co"} 0
....
```

To:
```
[root@node ~]# curl http://localhost:32221/metrics
....
lustre_stats_total{component="mdt",operation="sync",target="es02a-MDT0000"} 6
lustre_stats_total{component="mdt",operation="samedir_rename",target="es02a-MDT0000"} 2
lustre_stats_total{component="mdt",operation="parallel_rename_file",target="es02a-MDT0000"} 2

lustre_job_stats_total{operation="open",component="mdt",target="es01a-MDT0000",jobid="node_exporter:989:es-2-virt3-co"} 0
lustre_job_stats_total{operation="close",component="mdt",target="es01a-MDT0000",jobid="node_exporter:989:es-2-virt3-co"} 0
lustre_job_stats_total{operation="mknod",component="mdt",target="es01a-MDT0000",jobid="node_exporter:989:es-2-virt3-co"} 0
...
```

I haven't been able to reproduce it as part of unit test

### CODE REVIEW

File: lustrefs-exporter/src/jobstats.rs [hunk 1]
```diff
@@ -134,6 +134,9 @@ pub fn jobstats_stream<R: BufRead + std::marker::Send + 'static>(
     let x = tokio::task::spawn_blocking(move || {
         let mut state = State::Empty;
 
+        // Send a new line to avoid printing at the same level as the previous stats
+        _ = tx.blocking_send('\n'.to_compact_string());
```
Comment (breuhan) [inline lustrefs-exporter/src/jobstats.rs:138]: I'm wondering why we can't just send '\n' as char. I believe there is a reason, but I'm curious to learn why.

Comment (jgrund) [reply lustrefs-exporter/src/jobstats.rs:138]: We send [`compact_str`](https://docs.rs/compact_str/latest/compact_str/) as a profiling optimization over the channel (while carefully trying to keep strings within stack size).

**Review by breuhan — DISMISSED** [review]

File: lustrefs-exporter/src/jobstats.rs [hunk 2]
```diff
@@ -134,6 +134,9 @@ pub fn jobstats_stream<R: BufRead + std::marker::Send + 'static>(
     let x = tokio::task::spawn_blocking(move || {
         let mut state = State::Empty;
 
+        // Send a new line to make sure we are printing stats with a separating empty line
+        _ = tx.blocking_send("\n".to_compact_string());
```
Comment (jgrund) [inline lustrefs-exporter/src/jobstats.rs:138]: Is this needed? Do our other stats end with a newline char?

Comment (RDruon) [reply lustrefs-exporter/src/jobstats.rs:138]: It's not needed, it helps readability by adding a new line for jobstats related metric.

**Review by breuhan — APPROVED** [review]

**Review by jgrund — APPROVED** [review]

**Pull request merged**